protocol/bin/flow.js

```diff
@@ -28,20 +33,100 @@ function analyze (json) {
   return json.types.map(function (t) {
     switch (t.type) {
       case 'record':
-        return addRecord(json.protocol + '_', t)
+        return addRecord(`${json.protocol}_`, t)
       case 'enum':
-        return addEnum(json.protocol + '_', t)
+        return addEnum(`${json.protocol}_`, t)
       case 'fixed':
-        return addFixed(json.protocol + '_', t)
+        return addFixed(`${json.protocol}_`, t)
       default:
         return ''
     }
+  }).concat(analyzeMessages(json))
+}
+
+function figureKBType (protocol, rawType) {
+  return seenTypes[rawType] ? `${protocol}_${rawType}` : rawType
```

Comment:
this is just related to the function below where rawType comes in and type is us massaging it